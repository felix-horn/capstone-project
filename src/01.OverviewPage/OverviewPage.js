import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import ShopCard from './02.Components/ShopCard'
import ActionButton from './01.UI-Elements/ActionButton'
import ScannerButton from './01.UI-Elements/ScannerButton'

OverviewPage.propTypes = {
  database: PropTypes.object.isRequired,
  addShop: PropTypes.func.isRequired,
}

export default function OverviewPage({ database, addShop }) {
  return (
    <OverviewPageStyled>
      {database.shops.allIds.map((shopId) => (
        <ShopCard key={shopId} shopId={shopId} database={database} />
      ))}
      <ActionButtonStyled addShop={addShop} data-testid="acton-button" />
      <ScannerButtonStyled addShop={addShop} data-testid="acton-button" />
    </OverviewPageStyled>
  )
}

const OverviewPageStyled = styled.div`
  position: relative;
  height: calc(100vh - 60px);
  display: grid;
  gap: 10px;
  grid-auto-rows: minmax(min-content, max-content);
`

const ActionButtonStyled = styled(ActionButton)`
  position: absolute;
  bottom: 60px;
  right: 30px;
  z-index: 100;
`
const ScannerButtonStyled = styled(ScannerButton)`
  position: absolute;
  bottom: 120px;
  right: 30px;
  z-index: 100;
`
