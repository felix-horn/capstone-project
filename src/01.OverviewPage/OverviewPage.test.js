import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import { BrowserRouter as Router } from 'react-router-dom'
import OverviewPage from './OverviewPage'

const testDatabase = {
  shops: {
    allIds: ['x', 'y'],
    byId: {
      x: {
        id: 'x',
        title: 'shopX',
        items: ['a', 'b'],
      },
      y: {
        id: 'y',
        title: 'shopY',
        items: ['c'],
      },
    },
  },
  items: {
    allIds: ['a', 'b', 'c'],
    byId: {
      a: {
        id: 'a',
        title: 'itemA',
        isChecked: false,
      },
      b: {
        id: 'b',
        title: 'itemB',
        isChecked: true,
      },
      c: {
        id: 'c',
        title: 'itemC',
        isChecked: false,
      },
    },
  },
}

describe('OverviewPage', () => {
  it('renders correctly', () => {
    const addShopMock = jest.fn()
    const { container } = render(
      <Router>
        <OverviewPage database={testDatabase} addShop={addShopMock} />
      </Router>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls on addShop', () => {
    const addShopMock = jest.fn()
    const { getByTestId } = render(
      <Router>
        <OverviewPage database={testDatabase} addShop={addShopMock} />
      </Router>
    )
    const button = getByTestId('action-button')
    user.click(button)
    expect(addShopMock).toHaveBeenCalledTimes(1)
  })

  it('links to correct href', () => {
    const addShopMock = jest.fn()
    const { getByTestId } = render(
      <Router>
        <OverviewPage database={testDatabase} addShop={addShopMock} />
      </Router>
    )
    const button = getByTestId('action-button')
    user.click(button)
    expect(button).toHaveProperty('href', 'http://localhost/ShopPage')
  })
})
