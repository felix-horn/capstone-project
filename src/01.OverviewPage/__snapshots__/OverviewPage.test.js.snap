// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewPage renders correctly 1`] = `
.c2 {
  margin-bottom: 8px !important;
  font-size: 1.2rem;
  font-weight: 400;
  color: var(--almost-black);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5px;
}

.c4 {
  border: 1px solid var(--light-gray);
  border-radius: 2px;
  height: 15px;
  width: 15px;
}

.c5 {
  margin: 0;
  color: var(--dark-gray);
}

.c6 {
  margin: 4px 2px 0 !important;
  padding: 0;
  font-size: 0.8rem;
  color: var(--light-gray);
}

.c1 {
  border: var(--border);
  border-radius: 8px;
  padding: 15px;
  display: grid;
  gap: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  box-shadow: var(--box-shadow);
  border-radius: 100%;
  height: 50px;
  width: 50px;
  background-color: var(--white);
  opacity: 0.5;
  display: grid;
  place-items: center;
}

.c9 {
  color: var(--dark-gray);
}

.c10 {
  box-shadow: var(--box-shadow);
  border-radius: 100%;
  height: 50px;
  width: 50px;
  background-color: var(--white);
  opacity: 0.5;
  display: grid;
  place-items: center;
}

.c12 {
  color: var(--dark-gray);
}

.c0 {
  position: relative;
  height: calc(100vh - 60px);
  display: grid;
  gap: 10px;
  grid-auto-rows: minmax(min-content,max-content);
}

.c8 {
  position: absolute;
  bottom: 60px;
  right: 30px;
  z-index: 100;
}

.c11 {
  position: absolute;
  bottom: 120px;
  right: 30px;
  z-index: 100;
}

<div
  class="c0"
>
  <a
    class="c1"
    data-testid="shop-card"
    href="/ShopPage"
  >
    <h2
      class="c2"
    >
      shopX
    </h2>
    <label
      class="c3"
    >
      <div
        class="c4"
      />
      <p
        class="c5"
      >
        itemA
      </p>
    </label>
    <p
      class="c6"
    >
      und 1 abgehakter Eintrag
    </p>
  </a>
  <a
    class="c1"
    data-testid="shop-card"
    href="/ShopPage"
  >
    <h2
      class="c2"
    >
      shopY
    </h2>
    <label
      class="c3"
    >
      <div
        class="c4"
      />
      <p
        class="c5"
      >
        itemC
      </p>
    </label>
  </a>
  <a
    class="c7 c8"
    data-testid="action-button"
    href="/ShopPage"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root c9"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
      />
      <path
        d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
      />
    </svg>
  </a>
  <a
    class="c10 c11"
    data-testid="scanner-button"
    href="/ScannerPage"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root c12"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 5v4h2V5h4V3H5c-1.1 0-2 .9-2 2zm2 10H3v4c0 1.1.9 2 2 2h4v-2H5v-4zm14 4h-4v2h4c1.1 0 2-.9 2-2v-4h-2v4zm0-16h-4v2h4v4h2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </a>
</div>
`;
